### PR TITLE
Dev/sandeep

### PR DESCRIPTION
**Pull request by sandeepdhanni**
created camunda tasklist authentication token programatically and created api for forgot password

### CODE REVIEW

File: corsafe-poc-service/src/main/java/com/aus/corsafe/controller/ForgotPasswordController.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    @Autowired
+    private ForgotPasswordService forgotPasswordService;
+
+
+    //this api is save the forgot password
+    @PostMapping("/password/forgotPassword")
+    public ResponseEntity<?> forgotPassword(@RequestBody ForgotPassword forgotPassword) {
+        try {
+            String newPassword = forgotPasswordService.updatePassword(forgotPassword);
+            return new CommonResponse<>().prepareSuccessResponseObject("Password updated successfully",  HttpStatus.CREATED);
```
Comment (venkyEidiko) [inline corsafe-poc-service/src/main/java/com/aus/corsafe/controller/ForgotPasswordController.java:27]: Move the String constant into ApplicationConfig Class

Comment (venkyEidiko) [reply corsafe-poc-service/src/main/java/com/aus/corsafe/controller/ForgotPasswordController.java:27]: Create all String as a static String content and access as a **Appliacation.Variable_name**

File: corsafe-poc-service/src/main/java/com/aus/corsafe/service/ForgotPasswordService.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Autowired
+    private UserRegisterRepo userRegisterRepo;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    public String updatePassword(ForgotPassword forgotPassword) throws BadRequestException {
+        if (!forgotPassword.getNewPassword().equals(forgotPassword.getConfirmPassword())) {
+            throw new PasswordMismatchException("Passwords do not match");
```
Comment (venkyEidiko) [inline corsafe-poc-service/src/main/java/com/aus/corsafe/service/ForgotPasswordService.java:25]: Move it into ApplicationConfig Class

File: corsafe-poc-service/src/main/java/com/aus/corsafe/service/UserRegisterServiceImpl.java [hunk 3]
```diff
@@ -44,4 +47,21 @@ public List<SecurityQuestion> getAllSecurityQuestion() {
     }
 
 
+    @Override
+    public String findEmailByEmail(String email) {
+        Optional<UserRegister> user = userRegisterRepo.findByEmail(email);
+        if (user.isPresent()) {
+            return user.get().getEmail();
+        } else {
+            throw new BadCrediantialsCls("Email not registered with : " + email);
```
Comment (venkyEidiko) [inline corsafe-poc-service/src/main/java/com/aus/corsafe/service/UserRegisterServiceImpl.java:56]: MOve

File: corsafe-poc-service/src/main/java/com/aus/corsafe/service/bpmnservice/AuthService.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    private final WebClient webClient;
+
+    public AuthService(WebClient webClient) {
+        this.webClient = webClient;
+    }
+
+    public Mono<String> getAuthToken(String clientId, String clientSecret) {
+        String body = "grant_type=client_credentials" +
+                "&audience=tasklist.camunda.io" +
```
Comment (venkyEidiko) [inline corsafe-poc-service/src/main/java/com/aus/corsafe/service/bpmnservice/AuthService.java:20]: Read values from Yaml File

File: corsafe-poc-service/src/main/java/com/aus/corsafe/webClientConfig/WebClientConfig.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import org.springframework.web.reactive.function.client.WebClient;
+
+@Configuration
+public class WebClientConfig {
+
+
+    @Bean
+    public WebClient webClient() {
+        return WebClient.builder()
+                .baseUrl("https://login.cloud.camunda.io")
```
Comment (venkyEidiko) [inline corsafe-poc-service/src/main/java/com/aus/corsafe/webClientConfig/WebClientConfig.java:15]: Move it into Yaml file

File: corsafe-poc-service/src/main/java/com/aus/corsafe/service/bpmnservice/AuthService.java (hunk 4 above)
Comment (sandeepdhanni) [reply corsafe-poc-service/src/main/java/com/aus/corsafe/service/bpmnservice/AuthService.java:26]: i have resolved this issue..

**Pull request merged**